### PR TITLE
[minor] create student batch if does not exists

### PR DESCRIPTION
**Pull request by mbauskar**
```
Migrating itsc.erpnext.com
Executing erpnext.patches.v8_0.merge_student_batch_and_student_group in itsc.erpnext.com (c25d20e86e2dca4e)
{'program': u'Generl information about new vehicle type', 'academic_year': u'2016', 'student_group_name': u'C238-ITSC Driver-2016', 'batch': u'ITSC Driver', 'academic_term': None}
Traceback (most recent call last):
  File "/usr/lib64/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib64/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/frappe/benches/bench-2017-05-11/apps/frappe/frappe/utils/bench_helper.py", line 79, in <module>
    main()
  File "/home/frappe/benches/bench-2017-05-11/apps/frappe/frappe/utils/bench_helper.py", line 16, in main
    click.Group(commands=commands)(prog_name='bench')
  File "/home/frappe/benches/bench-2017-05-11/env/lib/python2.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/frappe/benches/bench-2017-05-11/env/lib/python2.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/frappe/benches/bench-2017-05-11/env/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/frappe/benches/bench-2017-05-11/env/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/frappe/benches/bench-2017-05-11/env/lib/python2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/frappe/benches/bench-2017-05-11/env/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/frappe/benches/bench-2017-05-11/env/lib/python2.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/frappe/benches/bench-2017-05-11/apps/frappe/frappe/commands/__init__.py", line 24, in _func
    ret = f(frappe._dict(ctx.obj), *args, **kwargs)
  File "/home/frappe/benches/bench-2017-05-11/apps/frappe/frappe/commands/site.py", line 214, in migrate
    migrate(context.verbose, rebuild_website=rebuild_website)
  File "/home/frappe/benches/bench-2017-05-11/apps/frappe/frappe/migrate.py", line 31, in migrate
    frappe.modules.patch_handler.run_all()
  File "/home/frappe/benches/bench-2017-05-11/apps/frappe/frappe/modules/patch_handler.py", line 29, in run_all
    if not run_single(patchmodule = patch):
  File "/home/frappe/benches/bench-2017-05-11/apps/frappe/frappe/modules/patch_handler.py", line 63, in run_single
    return execute_patch(patchmodule, method, methodargs)
  File "/home/frappe/benches/bench-2017-05-11/apps/frappe/frappe/modules/patch_handler.py", line 83, in execute_patch
    frappe.get_attr(patchmodule.split()[0] + ".execute")()
  File "/home/frappe/benches/bench-2017-05-11/apps/erpnext/erpnext/patches/v8_0/merge_student_batch_and_student_group.py", line 32, in execute
    doc.save()
  File "/home/frappe/benches/bench-2017-05-11/apps/frappe/frappe/model/document.py", line 230, in save
    return self._save(*args, **kwargs)
  File "/home/frappe/benches/bench-2017-05-11/apps/frappe/frappe/model/document.py", line 253, in _save
    self.insert()
  File "/home/frappe/benches/bench-2017-05-11/apps/frappe/frappe/model/document.py", line 193, in insert
    self._validate()
  File "/home/frappe/benches/bench-2017-05-11/apps/frappe/frappe/model/document.py", line 400, in _validate
    self._validate_links()
  File "/home/frappe/benches/bench-2017-05-11/apps/frappe/frappe/model/document.py", line 635, in _validate_links
    frappe.LinkValidationError)
  File "/home/frappe/benches/bench-2017-05-11/apps/frappe/frappe/__init__.py", line 316, in throw
    msgprint(msg, raise_exception=exc, title=title, indicator='red')
  File "/home/frappe/benches/bench-2017-05-11/apps/frappe/frappe/__init__.py", line 306, in msgprint
    _raise_exception()
  File "/home/frappe/benches/bench-2017-05-11/apps/frappe/frappe/__init__.py", line 279, in _raise_exception
    raise raise_exception, encode(msg)
frappe.exceptions.LinkValidationError: Could not find Batch: ITSC Driver
```

### CODE REVIEW

File: erpnext/patches/v8_0/merge_student_batch_and_student_group.py [hunk 1]
```diff
@@ -13,7 +13,15 @@ def execute():
 	frappe.reload_doctype("Student Group")
 	student_batches = frappe.db.sql('''select name as student_group_name, student_batch_name as batch,
 		program, academic_year, academic_term from `tabStudent Batch`''', as_dict=1)
+
 	for student_batch in student_batches:
+		# create student batch name if does not exists !!
+		if not frappe.db.exists("Student Batch Name", student_batch.get("batch")):
```
Comment (nabinhait) [inline erpnext/patches/v8_0/merge_student_batch_and_student_group.py]: if student_batch.get("batch") and not frappe.db.exists("Student Batch Name", student_batch.get("batch")):

**Review by nabinhait — CHANGES_REQUESTED** [review]

**Comment by Helkyd** [comment]
Another error for Student
Executing erpnext.patches.v8_0.merge_student_batch_and_student_group in demo.angolaerp (4a8671fcb4e4542a)
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/helio/frappe-bench/apps/frappe/frappe/utils/bench_helper.py", line 79, in <module>
    main()
  File "/home/helio/frappe-bench/apps/frappe/frappe/utils/bench_helper.py", line 16, in main
    click.Group(commands=commands)(prog_name='bench')
  File "/home/helio/frappe-bench/env/local/lib/python2.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/helio/frappe-bench/env/local/lib/python2.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/helio/frappe-bench/env/local/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/helio/frappe-bench/env/local/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/helio/frappe-bench/env/local/lib/python2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/helio/frappe-bench/env/local/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/helio/frappe-bench/env/local/lib/python2.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/helio/frappe-bench/apps/frappe/frappe/commands/__init__.py", line 24, in _func
    ret = f(frappe._dict(ctx.obj), *args, **kwargs)
  File "/home/helio/frappe-bench/apps/frappe/frappe/commands/site.py", line 214, in migrate
    migrate(context.verbose, rebuild_website=rebuild_website)
  File "/home/helio/frappe-bench/apps/frappe/frappe/migrate.py", line 31, in migrate
    frappe.modules.patch_handler.run_all()
  File "/home/helio/frappe-bench/apps/frappe/frappe/modules/patch_handler.py", line 29, in run_all
    if not run_single(patchmodule = patch):
  File "/home/helio/frappe-bench/apps/frappe/frappe/modules/patch_handler.py", line 63, in run_single
    return execute_patch(patchmodule, method, methodargs)
  File "/home/helio/frappe-bench/apps/frappe/frappe/modules/patch_handler.py", line 83, in execute_patch
    frappe.get_attr(patchmodule.split()[0] + ".execute")()
  File "/home/helio/frappe-bench/apps/erpnext/erpnext/patches/v8_0/merge_student_batch_and_student_group.py", line 28, in execute
    where parent=%s''', (doc.name), as_dict=1)
  File "/home/helio/frappe-bench/apps/frappe/frappe/database.py", line 138, in sql
    self._cursor.execute(query, values)
  File "/home/helio/frappe-bench/env/local/lib/python2.7/site-packages/MySQLdb/cursors.py", line 205, in execute
    self.errorhandler(self, exc, value)
  File "/home/helio/frappe-bench/env/local/lib/python2.7/site-packages/MySQLdb/connections.py", line 36, in defaulterrorhandler
    raise errorclass, errorvalue
_mysql_exceptions.OperationalError: (1054, "Unknown column 'active' in 'field list'")

**Comment by Helkyd** [comment]
After manually adding the field ACTIVE i ran update again and this time the error is on
student_batch.update({"doctype":"Student Group", "group_based_on": "Batch"})
below the error trace

Executing erpnext.patches.v8_0.merge_student_batch_and_student_group in demo.angolaerp (4a8671fcb4e4542a)
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/helio/frappe-bench/apps/frappe/frappe/utils/bench_helper.py", line 79, in <module>
    main()
  File "/home/helio/frappe-bench/apps/frappe/frappe/utils/bench_helper.py", line 16, in main
    click.Group(commands=commands)(prog_name='bench')
  File "/home/helio/frappe-bench/env/local/lib/python2.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/helio/frappe-bench/env/local/lib/python2.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/helio/frappe-bench/env/local/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/helio/frappe-bench/env/local/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/helio/frappe-bench/env/local/lib/python2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/helio/frappe-bench/env/local/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/helio/frappe-bench/env/local/lib/python2.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/helio/frappe-bench/apps/frappe/frappe/commands/__init__.py", line 24, in _func
    ret = f(frappe._dict(ctx.obj), *args, **kwargs)
  File "/home/helio/frappe-bench/apps/frappe/frappe/commands/site.py", line 214, in migrate
    migrate(context.verbose, rebuild_website=rebuild_website)
  File "/home/helio/frappe-bench/apps/frappe/frappe/migrate.py", line 31, in migrate
    frappe.modules.patch_handler.run_all()
  File "/home/helio/frappe-bench/apps/frappe/frappe/modules/patch_handler.py", line 29, in run_all
    if not run_single(patchmodule = patch):
  File "/home/helio/frappe-bench/apps/frappe/frappe/modules/patch_handler.py", line 63, in run_single
    return execute_patch(patchmodule, method, methodargs)
  File "/home/helio/frappe-bench/apps/frappe/frappe/modules/patch_handler.py", line 83, in execute_patch
    frappe.get_attr(patchmodule.split()[0] + ".execute")()
  File "/home/helio/frappe-bench/apps/erpnext/erpnext/patches/v8_0/merge_student_batch_and_student_group.py", line 39, in execute
    doc.save()
  File "/home/helio/frappe-bench/apps/frappe/frappe/model/document.py", line 230, in save
    return self._save(*args, **kwargs)
  File "/home/helio/frappe-bench/apps/frappe/frappe/model/document.py", line 253, in _save
    self.insert()
  File "/home/helio/frappe-bench/apps/frappe/frappe/model/document.py", line 189, in insert
    self.validate_higher_perm_levels()
  File "/home/helio/frappe-bench/apps/frappe/frappe/model/document.py", line 458, in validate_higher_perm_levels
    high_permlevel_fields = frappe.get_meta(df.options).meta.get_high_permlevel_fields()
  File "/home/helio/frappe-bench/apps/frappe/frappe/__init__.py", line 623, in get_meta
    return frappe.model.meta.get_meta(doctype, cached=cached)
  File "/home/helio/frappe-bench/apps/frappe/frappe/model/meta.py", line 31, in get_meta
    return frappe.cache().hget("meta", doctype, lambda: Meta(doctype))
  File "/home/helio/frappe-bench/apps/frappe/frappe/utils/redis_wrapper.py", line 171, in hget
    value = generator()
  File "/home/helio/frappe-bench/apps/frappe/frappe/model/meta.py", line 31, in <lambda>
    return frappe.cache().hget("meta", doctype, lambda: Meta(doctype))
  File "/home/helio/frappe-bench/apps/frappe/frappe/model/meta.py", line 66, in __init__
    super(Meta, self).__init__("DocType", doctype)
  File "/home/helio/frappe-bench/apps/frappe/frappe/model/document.py", line 84, in __init__
    self.load_from_db()
  File "/home/helio/frappe-bench/apps/frappe/frappe/model/meta.py", line 71, in load_from_db
    super(Meta, self).load_from_db()
  File "/home/helio/frappe-bench/apps/frappe/frappe/model/document.py", line 115, in load_from_db
    frappe.throw(_("{0} {1} not found").format(_(self.doctype), self.name), frappe.DoesNotExistError)
  File "/home/helio/frappe-bench/apps/frappe/frappe/__init__.py", line 316, in throw
    msgprint(msg, raise_exception=exc, title=title, indicator='red')
  File "/home/helio/frappe-bench/apps/frappe/frappe/__init__.py", line 306, in msgprint
    _raise_exception()
  File "/home/helio/frappe-bench/apps/frappe/frappe/__init__.py", line 279, in _raise_exception
    raise raise_exception, encode(msg)
frappe.exceptions.DoesNotExistError: DocType Student Group Instructor not found

**Comment by nabinhait** [comment]
sorry, will release the fix in a hour

**Pull request merged**